### PR TITLE
Check for edge cases

### PR DESCRIPTION
**Pull request by niklashtlwrn**
Changes:
- Check for edge cases in the concatenate.py file.
- Add tests for the edge cases in the test_concatenate.py file.

### CODE REVIEW

File: concatenate.py [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
     """Concatenate a list of strings into a single string."""
+    
+    # check if list_of_strings contains only strings
+    for string in list_of_strings:
+        if not isinstance(string, str):
+            raise TypeError('list_of_strings must contain only strings!')
+        
+    # check for length of list_of_strings
+    if list_of_strings == []:
+        return ''
```
Comment (grzegorz-karas) [inline concatenate.py:11]: this condition is unnecessary because:
```
>>> ' '.join([])
''
```

File: concatenate.py [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+    elif len(list_of_strings) == 1:
+        return list_of_strings[0]
```
Comment (grzegorz-karas) [inline concatenate.py:13]: this condition is unnecessary because:
```
>>> ' '.join(['string_1'])
'string_1'
```

File: test_concatenate.py [hunk 3]
```diff
@@ -2,4 +2,21 @@
...
+    assert 'string_1' == concatenate(list_of_strings)
+
+def test_concatenate_raises_type_error():
+    list_of_strings = ['string_1', 1]
+    try:
+        concatenate(list_of_strings)
+    except TypeError:
+        assert True
+    else:
+        assert False
```
Comment (grzegorz-karas) [inline test_concatenate.py:22]: Would you please group the test as described [here](https://docs.pytest.org/en/7.1.x/getting-started.html#group-multiple-tests-in-a-class)?

**Review by grzegorz-karas — CHANGES_REQUESTED** [review]
Less code is better than more code. Would you please take a look into my comments?

**Review by grzegorz-karas — APPROVED** [review]
LGTM
